src/ssl.c

```diff
@@ -22991,7 +22991,13 @@ int wolfSSL_i2d_PublicKey(const WOLFSSL_EVP_PKEY *key, unsigned char **der)
     if (ret == 0) {
         local_derSz = wolfSSL_EVP_PKEY_get_der(key, &local_der);
         if (local_derSz <= 0) {
-            ret = WOLFSSL_FATAL_ERROR;
+            /* In this case, there was no buffered DER at all. This could be the
+             * case where the key that was passed in was generated. So now we
+             * have to create the local DER. */
+            local_derSz = wolfSSL_i2d_ECPrivateKey(key->ecc, &local_der);
```

Comment:
Farther down `local_derSz` gets used in `wc_EccPublicKeyDecode`. Consider an explicit cast to `word32` avoiding future warnings.

Reply:
Changed back to word32.  The trouble was that it can return -1 so it becomes max value of word32.  I have a different solution. Please stay tuned.